sources/packages/backend/apps/db-migrations/src/sql/EducationProgramsOfferings/Update-offering-duplication-unique-index.sql

```diff
@@ -0,0 +1,26 @@
+-- Commits the addition of the new offering_type 'Scholastic standing'
+commit;
```

Comment:
Unless I am missing something this solution would not be desirable or we need to at least discuss it as a team.
If my understanding is correct it is actually committing the parent transaction created by typerom to execute all pending SQL migrations.
I am assuming that the error happens when the database is built from scratch to execute, for instance, the E2E tests.
We can change the typeorm to execute the transaction per migration for E2E tests (`npx typeorm migration:run -t each`) and leave the "PROD" migrations as they are right now unless DEVs would be willing to change it all to be execute with the `each` mode. Last time that it was proposed the decision was not to use the `each` for PROD.
@dheepak-aot @guru-aot @sh16011993 @lewischen-aot thoughts?

Reply:
why should the index be targeting only 'Public' offerings?

Reply:
Because they are the ones where the duplication would cause some impact and also the ones imported during the bulk upload. Private are student-specific, and, unless businesses argue against it, I believe that the institutions must have the ability to create duplicated ones to satisfy a PIR.
@HRAGANBC @CarlyCotton @Joshua-Lakusta thoughts?

Reply:
i can see the problem here, I am okay to change the execute transaction -t each in e2e tests. If that is the case as per my understanding, we have to taken into account that every migration will be taking a separate transaction so if one migration fails on the way, the rollback will happen for that particular transaction and the db will be in a half migrated state.  Are we okay for this?

Reply:
I agree with @andrewsignori-aot and I strongly believe that

 - First of all, we should do that commit manually in migrations
 - Second, we can modify the query and move forward
 - Third, same issue has happened before. I can see this thread of discussion and we can discuss this broadly in next dev sync up.

![image](https://github.com/bcgov/SIMS/assets/54600590/35efe700-3c68-4f61-ae1e-f3df93e47206)

Reply:
I'm still learning the system, and I can see the problem with the bulk upload indexing or id. I'm just thinking if it would be too much work for having two columns to map each item during bulk upload. One column stores the indexing from bulk upload, the other column stores a unique index from uuid or something. It's a thought here. I could be totally wrong.

Reply:
> i can see the problem here, I am okay to change the execute transaction -t each in e2e tests. If that is the case as per my understanding, we have to taken into account that every migration will be taking a separate transaction so if one migration fails on the way, the rollback will happen for that particular transaction and the db will be in a half migrated state. Are we okay for this?

@guru-aot for E2E tests I do not see a problem with that. The concern arises for Staging/PROD deployments only, IMO.

Reply:
Institutions don't need to create duplicate offerings to satisfy a PIR. They aren't expecting to create them as they need some sort of indicator on their end to identify how/why these are different offerings. We've provided tips in their instructional guide on how to create them during the PIR process.
If they're private and can be duplicated there is possibility for confusion on the Institution side.

Reply:
Thanks @CarlyCotton for the inputs. As per the team's decision, we will take that into consideration and change the code as suggested.
We will be using `AND offering_type IN ('Public', 'Private');`